CI build job lyrical/main failed!

An automated scheduled build failed on `master`: https://github.com/ros-controls/ros2_controllers/actions/runs/28421428813

```
2026-06-30T05:16:39.2310930Z Starting >>> rqt_joint_trajectory_controller
2026-06-30T05:16:39.2311242Z --- output: rqt_joint_trajectory_controller
2026-06-30T05:16:39.2311773Z [1m============================= test session starts ==============================[0m
2026-06-30T05:16:39.2312378Z platform linux -- Python 3.14.4, pytest-9.0.2, pluggy-1.6.0
2026-06-30T05:16:39.2313191Z cachedir: /home/runner/work/ros2_controllers/ros2_controllers/.work/target_ws/build/rqt_joint_trajectory_controller/.pytest_cache
2026-06-30T05:16:39.2314164Z rootdir: /home/runner/work/ros2_controllers/ros2_controllers/.work/target_ws/src/ros2_controllers/rqt_joint_trajectory_controller
2026-06-30T05:16:39.2315651Z plugins: ament_copyright-0.20.6, ament_xmllint-0.20.6, ament_mypy-0.20.6, launch_testing-3.9.7, launch_testing_ros-0.29.8, ament_lint-0.20.6, ament_flake8-0.20.6, ament_pep257-0.20.6, typeguard-4.4.4, repeat-0.9.4, cov-5.0.0, colcon-core-0.21.0, rerunfailures-16.1
2026-06-30T05:16:39.2316951Z [1mcollecting ... [0m[1m
2026-06-30T05:16:39.2317346Z collected 23 items                                                             [0m
2026-06-30T05:16:39.2318108Z 
2026-06-30T05:16:39.2319101Z test/test_joint_limits_urdf.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m                    [ 95%][0m
2026-06-30T05:16:39.2320302Z test/test_run_launch.py [31mF[0m[31m                                                [100%][0m
2026-06-30T05:16:39.2320742Z 
2026-06-30T05:16:39.2320918Z =================================== FAILURES ===================================
2026-06-30T05:16:39.2321445Z [31m[1m______________________ launch tests: test.test_run_launch ______________________[0m
2026-06-30T05:16:39.2321861Z 
2026-06-30T05:16:39.2322181Z ======================================================================================================================================================================================
2026-06-30T05:16:39.2322687Z FAIL: test.test_run_launch.TestShutdown.test_exit_codes
2026-06-30T05:16:39.2323487Z --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
2026-06-30T05:16:39.2324132Z Traceback (most recent call last):
2026-06-30T05:16:39.2325207Z   File "/home/runner/work/ros2_controllers/ros2_controllers/.work/target_ws/src/ros2_controllers/rqt_joint_trajectory_controller/test/test_run_launch.py", line 75, in test_exit_codes
2026-06-30T05:16:39.2326192Z     launch_testing.asserts.assertExitCodes(
2026-06-30T05:16:39.2326558Z     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
2026-06-30T05:16:39.2326964Z         proc_info, allowable_exit_codes=allowable_exit_codes
2026-06-30T05:16:39.2327297Z         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-06-30T05:16:39.2327777Z     )
2026-06-30T05:16:39.2328127Z     ^
2026-06-30T05:16:39.2328643Z   File "/opt/ros/lyrical/lib/python3.14/site-packages/launch_testing/asserts/assert_exit_codes.py", line 62, in assertExitCodes
2026-06-30T05:16:39.2329553Z     assert info.returncode in allowable_exit_codes, 'Proc {} exited with code {}'.format(
2026-06-30T05:16:39.2330006Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-06-30T05:16:39.2330562Z AssertionError: Proc rqt_joint_trajectory_controller-1 exited with code -11
2026-06-30T05:16:39.2331060Z ----------------------------- Captured stdout call -----------------------------
2026-06-30T05:16:39.2331765Z [INFO] [launch]: All log files can be found below /root/.ros/log/2026-06-30-05-16-35-079201-9e4923c3343c-16287
2026-06-30T05:16:39.2332283Z [INFO] [launch]: Default logging verbosity is set to INFO
2026-06-30T05:16:39.2332855Z [INFO] [rqt_joint_trajectory_controller-1]: process started with pid [16299]
2026-06-30T05:16:39.2335010Z [rqt_joint_trajectory_controller-1] [WARN] [1782796596.057620184] [rcl.logging_rosout]: Publisher already registered for node name: 'rqt_joint_trajectory_controller_node'. If this is due to multiple nodes with the same name then all logs for the logger named 'rqt_joint_trajectory_controller_node' will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
2026-06-30T05:16:39.2338953Z [rqt_joint_trajectory_controller-1] [WARN] [1782796596.095733435] [rcl.logging_rosout]: Publisher already registered for node name: 'rqt_joint_trajectory_controller_node'. If this is due to multiple nodes with the same name then all logs for the logger named 'rqt_joint_trajectory_controller_node' will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
2026-06-30T05:16:39.2341284Z [rqt_joint_trajectory_controller-1] [36mThis plugin does not support propagateSizeHints()[0m
2026-06-30T05:16:39.2341977Z [INFO] [rqt_joint_trajectory_controller-1]: sending signal 'SIGINT' to process[rqt_joint_trajectory_controller-1]
2026-06-30T05:16:39.2342728Z [rqt_joint_trajectory_controller-1] [36mThis plugin does not support propagateSizeHints()[0m
2026-06-30T05:16:39.2344166Z [ERROR] [rqt_joint_trajectory_controller-1]: process has died [pid 16299, exit code -11, cmd '/home/runner/work/ros2_controllers/ros2_controllers/.work/target_ws/install/rqt_joint_trajectory_controller/lib/rqt_joint_trajectory_controller/rqt_joint_trajectory_controller --ros-args -r __node:=rqt_joint_trajectory_controller_node'].
2026-06-30T05:16:39.2345648Z ----------------------------- Captured stderr call -----------------------------
2026-06-30T05:16:39.2346127Z test_node_start (test.test_run_launch.TestFixture.test_node_start) ... ok
2026-06-30T05:16:39.2346417Z 
2026-06-30T05:16:39.2346567Z ----------------------------------------------------------------------
2026-06-30T05:16:39.2346893Z Ran 1 test in 2.049s
2026-06-30T05:16:39.2347021Z 
2026-06-30T05:16:39.2347107Z OK
2026-06-30T05:16:39.2347383Z test_exit_codes (test.test_run_launch.TestShutdown.test_exit_codes)
2026-06-30T05:16:39.2348032Z Check if the process exited normally. ... FAIL
2026-06-30T05:16:39.2348248Z 
2026-06-30T05:16:39.2348359Z ======================================================================
2026-06-30T05:16:39.2348879Z FAIL: test_exit_codes (test.test_run_launch.TestShutdown.test_exit_codes)
2026-06-30T05:16:39.2349291Z Check if the process exited normally.
2026-06-30T05:16:39.6935632Z ----------------------------------------------------------------------
2026-06-30T05:16:39.6936468Z Traceback (most recent call last):
2026-06-30T05:16:39.6939247Z   File "/home/runner/work/ros2_controllers/ros2_controllers/.work/target_ws/src/ros2_controllers/rqt_joint_trajectory_controller/test/test_run_launch.py", line 75, in test_exit_codes
2026-06-30T05:16:39.6940983Z     launch_testing.asserts.assertExitCodes(
2026-06-30T05:16:39.6941525Z     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
2026-06-30T05:16:39.6942101Z         proc_info, allowable_exit_codes=allowable_exit_codes
2026-06-30T05:16:39.6942683Z         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-06-30T05:16:39.6943155Z     )
2026-06-30T05:16:39.6943443Z     ^
2026-06-30T05:16:39.6944304Z   File "/opt/ros/lyrical/lib/python3.14/site-packages/launch_testing/asserts/assert_exit_codes.py", line 62, in assertExitCodes
2026-06-30T05:16:39.6945642Z     assert info.returncode in allowable_exit_codes, 'Proc {} exited with code {}'.format(
2026-06-30T05:16:39.6946457Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-06-30T05:16:39.6947202Z AssertionError: Proc rqt_joint_trajectory_controller-1 exited with code -11
```

--- a/rqt_joint_trajectory_controller/rqt_joint_trajectory_controller/joint_trajectory_controller.py
+++ b/rqt_joint_trajectory_controller/rqt_joint_trajectory_controller/joint_trajectory_controller.py
@@ -203,6 +203,17 @@ class JointTrajectoryController(Plugin):
         self._unregister_state_sub()
         self._unregister_cmd_pub()
         self._unregister_executor()
+        self._destroy_node()
+
+    def _destroy_node(self):
+        if self._node is not None:
+            try:
+                self._node.destroy_node()
+            except Exception as e:
+                if not is_shutdown_context_error(e):
+                    raise
+            finally:
+                self._node = None
 
     def save_settings(self, plugin_settings, instance_settings):
         instance_settings.set_value("cm_ns", self._cm_ns)

